net.go

```diff
@@ -10,17 +10,26 @@ import (
 	"fmt"
 )
 
+type NICConfiguration struct {
+	DHCPenabled bool   `json:"dhcp_enabled"`
+	Gateway     string `json:"gateway"`
+	IPv4        string `json:"ipv4"`
+	IPv6        string `json:"ipv6"`
+}
```

Comment:
Please move the changes adding this new NICConfiguration to a separate pull request where we can discuss this feature in isolation from the Windows-specific patch here.